runner/src/main.rs

```diff
@@ -437,18 +730,53 @@ fn workspace_manifest_files() -> impl Iterator<Item = PathBuf> {
         .filter(is_cargo_workspace_file)
 }
 
+/// Return whether the provided path refers to a source file in a programming language.
+fn is_source_code_file(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename.ends_with(".cc")
+        || filename.ends_with(".h")
+        || filename.ends_with(".rs")
+        || filename.ends_with(".proto")
+}
+
+/// Return whether the provided path refers to a source file that can be formatted by clang-tidy.
+fn is_clang_format_file(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename.ends_with(".cc") || filename.ends_with(".h") || filename.ends_with(".proto")
+}
+
 /// Return whether the provided path refers to a Bazel file (`BUILD`, `WORKSPACE`, or `*.bzl`)
 fn is_bazel_file(path: &PathBuf) -> bool {
     let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
     filename == "BUILD" || filename == "WORKSPACE" || filename.ends_with(".bzl")
 }
 
+fn is_build_file(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename == "BUILD"
+}
+
 /// Return whether the provided path refers to a markdown file (`*.md`)
 fn is_markdown_file(path: &PathBuf) -> bool {
     let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
     filename.ends_with(".md")
 }
 
+fn is_dockerfile(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename.ends_with("Dockerfile")
+}
+
+fn is_toml_file(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename.ends_with(".toml")
+}
+
+fn is_yaml_file(path: &PathBuf) -> bool {
+    let filename = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+    filename.ends_with(".yaml")
+}
```

Comment:
I might be overengineering things, but since these are quite common, do you think it would make sense to abstract this logic with a higher-order-function?

Eg a function that accepts a path, returns a function that then accepts regexes that check the filename. Would make things DRYer, but also maybe more complex. 

Could then be used like so 

```
.filter(match_file_types(yaml_regex, toml_regex, markdown_regex))
```

Reply:
Haha, I see. :D I think it's technically outside of the scope of the work I pitched to do on Oak, but will give it a shot for the fun of it.